src/components/TupleSelect/TupleSelect.test.tsx

```diff
@@ -0,0 +1,176 @@
+  expect(input).toHaveValue("");
+});
+
+test("Should clear input when a value is submitted", () => {
+  const { queryByDataCy } = render(Content, {
+    route: `/commits/evergreen`,
+    path: "/commits/:projectId",
+  });
+  const input = queryByDataCy("tuple-select-input");
+
+  expect(input).toHaveValue("");
+  fireEvent.change(input, {
+    target: { value: "some-filter" },
+  });
+  expect(input).toHaveValue("some-filter");
+  fireEvent.focus(input);
+  fireEvent.keyDown(input, {
+    key: "Enter",
+    keyCode: 13,
+  });
+  expect(input).toHaveValue("");
+});
+
+test("Should add input query params to the url", () => {
+  const { queryByDataCy, history } = render(Content, {
+    route: `/commits/evergreen`,
+    path: "/commits/:projectId",
+  });
+  const input = queryByDataCy("tuple-select-input");
+
+  expect(input).toHaveValue("");
+  fireEvent.change(input, {
+    target: { value: "some-filter" },
+  });
+  expect(input).toHaveValue("some-filter");
+  fireEvent.focus(input);
+  fireEvent.keyDown(input, {
+    key: "Enter",
+    keyCode: 13,
+  });
+  const { location } = history;
+
+  expect(location.search).toBe(
+    `?${ProjectFilterOptions.BuildVariant}=some-filter`
+  );
+});
+
+test("Should add multiple input filters to the same key as query params", () => {
+  const { queryByDataCy, history } = render(Content, {
+    route: `/commits/evergreen`,
+    path: "/commits/:projectId",
+  });
+  const input = queryByDataCy("tuple-select-input");
+  expect(input).toHaveValue("");
+  fireEvent.change(input, {
+    target: { value: "some-filter" },
+  });
+  expect(input).toHaveValue("some-filter");
+  fireEvent.keyDown(input, {
+    key: "Enter",
+    keyCode: 13,
+  });
+  fireEvent.change(input, {
+    target: { value: "some-other-filter" },
+  });
+  expect(input).toHaveValue("some-other-filter");
+  fireEvent.keyDown(input, {
+    key: "Enter",
+    keyCode: 13,
+  });
+  const { location } = history;
+  expect(location.search).toBe(
+    `?${ProjectFilterOptions.BuildVariant}=some-filter,some-other-filter`
+  );
+});
+
+test("Should not allow duplicate input filters for the same key as query params", () => {
+  const { queryByDataCy, history } = render(Content, {
+    route: `/commits/evergreen`,
+    path: "/commits/:projectId",
+  });
+  const input = queryByDataCy("tuple-select-input");
+  expect(input).toHaveValue("");
+  fireEvent.change(input, {
+    target: { value: "some-filter" },
+  });
+  expect(input).toHaveValue("some-filter");
+  fireEvent.keyDown(input, {
+    key: "Enter",
+    keyCode: 13,
+  });
+  fireEvent.change(input, {
+    target: { value: "some-filter" },
+  });
+  expect(input).toHaveValue("some-filter");
+  fireEvent.keyDown(input, {
+    key: "Enter",
+    keyCode: 13,
+  });
+  const { location } = history;
+  expect(location.search).toBe(
+    `?${ProjectFilterOptions.BuildVariant}=some-filter`
+  );
+});
+
+// test("Should allow multiple input filters for different keys as query params", () => {
```

Comment:
I'm still trying to write this test but theres an issue with the antd select component and react-testing-library. 
https://stackoverflow.com/questions/61080116/ant-design-v4-breaks-react-testing-library-tests-for-select-and-autocomplete